### PR TITLE
feat: add Ory Kratos authentication scripts

### PR DESCRIPTION
**Pull request by getlarge**
## Description

I added two authentication scripts to enable authenticated interacted with web applications relying on [Ory Kratos](https://github.com/ory/kratos).
- `KratosApiAuthentication.js` adds support for the [Native Login flow](https://www.ory.sh/docs/kratos/self-service/flows/user-login#login-for-api-clients-and-clients-without-browsers)
- `KratosABrowserAuthentication.js` adds support for the [Browser Login flow](https://www.ory.sh/docs/kratos/self-service/flows/user-login#login-for-server-side-browser-clients)

I hope they may be useful for others in the Zap community.

### CODE REVIEW

**Comment by thc202** [comment]
The commits need to be signed-off: https://github.com/zaproxy/community-scripts/pull/447/checks?check_run_id=25640107047
You need to run the `spotlessApply` task (e.g. `./gradlew sA`) to fix the formatting of the scripts. 
The changelog should be updated to mention the new scripts: https://github.com/zaproxy/community-scripts/blob/main/CONTRIBUTING.md#changelog

**Comment by getlarge** [comment]
> The commits need to be signed-off: https://github.com/zaproxy/community-scripts/pull/447/checks?check_run_id=25640107047 You need to run the `spotlessApply` task (e.g. `./gradlew sA`) to fix the formatting of the scripts. The changelog should be updated to mention the new scripts: https://github.com/zaproxy/community-scripts/blob/main/CONTRIBUTING.md#changelog

Sorry, I missed the contribution guidelines. Too bad I had to remove my GPG signature ;)
Should be good now!

**Comment by getlarge** [comment]
Yes, but not during the rebase unfortunately.

**Comment by getlarge** [comment]
@thc202 Can you help for the reviewing of this PR? Is anything blocking?

File: authentication/KratosApiAuthentication.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
+/*
+ * This script is used to authenticate a user using Ory Kratos self-service API for clients without browsers.
+ *
+ * Authentication Verification can be configured with the following rule:
+ * - Verification Strategy: Poll the Specified URL
+ * - Regex Pattern used to identify Logged In message: \Qactive.*
```
Comment (thc202) [inline authentication/KratosApiAuthentication.js:6]: This pattern does not look right, missing `\E` before `.*`? Or remove `\Q`? (Some of the comments added in this script also apply to the other.)

File: authentication/KratosApiAuthentication.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+ * - URL to poll: <Kratos Base URL>/sessions/whoami
+ *
+ * Zap must be configured to use HTTP header-based session management with the value: {%json:session_token%}
```
Comment (thc202) [inline authentication/KratosApiAuthentication.js:9]: ZAP

File: authentication/KratosApiAuthentication.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ * @returns {Object} The HTTP message used to perform the authentication.
+ */
+function authenticate(helper, paramsValues, credentials) {
+  print("Authenticating via Ory Kratos...");
+
+  // Step 1: Initialize the login flow
+  const kratosBaseUri = paramsValues.get("Kratos Base URL");
+  const initLoginUri = new URI(
+    kratosBaseUri + "/self-service/login/api",
+    false
```
Comment (thc202) [inline authentication/KratosApiAuthentication.js:54]: Would be better if this was true, having the URI already escaped/encoded avoids surprises.

Comment (getlarge) [reply authentication/KratosApiAuthentication.js:54]: Honestly here I just took inspiration from the other Authentication snippets which all seems to set false as a value for the second parameter.
If I understand correctly, you would suggest to do something like:
```js
new URI(encodeURI(kratosBaseUri + "/self-service/login/api"), true);
```
for all URI instantiation?

Comment (thc202) [reply authentication/KratosApiAuthentication.js:54]: Yes.

File: authentication/KratosApiAuthentication.js (hunk 1 above)
Comment (getlarge) [reply authentication/KratosApiAuthentication.js:6]: Oh yes, thanks for pointing this out! That makes the tests much faster ;)

**Comment by getlarge** [comment]
@thc202 Just applied your suggestions.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**